.circleci/config.yml

```diff
@@ -188,7 +188,7 @@ jobs:
       name: cl2-back
       image-tag: $CIRCLE_SHA1
     working_directory: /cl2_back
-    parallelism: 4
+    parallelism: 8
```

Comment:
I know I wrote this in the ticket, but I'm not wondering whether it's worth the extra $$$, maybe 5 or 6 is fine too?

The startup cost starts to add up (everything before the actual tests are executing) when parallelism is really high, so maybe we should make it depend on that time. @sebastienhoorens  is still working on reducing git pull time and @adessy on slightly reducing the image size, so we can expect it to go down a little.

Could you make a rough estimate of how the cost/speed compares 4-8 parallelism? You can find the credits/min price [here](https://circleci.com/product/features/resource-classes/).

Reply:
Comparing the runs before and after, the time of the checkout step was reduced from 29s to 4s.

Reply:
> the time of the checkout step was reduced from 29s to 4s

Good job! Thank you! Is it already in master?

Reply:
Not yet, I'm trying to apply this in as many parts of our workflows as possible in my branch.

Reply:
That's great! For this specific step,however, it doesn't really apply. There is no checkout, that happens in the build step before.

However, @alexander-cit , in your work of optimizing the e2e tests, if you're exploring running db:load in a separate job before the actual tests and transferring the pgdata, that technique could also make a difference for `back-test` to reduce startup costs. Not sure it's worth the extra complexity though.

Reply:
> I'm trying to apply this in as many parts of our workflows as possible in my branch.

You talking about my work (this PR) or your work?

Reply:
`db:load` is pretty fast: ~8 seconds for `docker-compose run --rm -e RAILS_ENV=test web bin/rails db:drop db:create db:schema:load`, so probably not worth it.

Reply:
@kogre I'll do 5 runs at `parallelism: 6` & 5 at `8`. There's a lot of variance in performance, so this seems to be the minimum to get any kind of accuracy.

Reply:
@Kogre The results are in!

I didn’t test less than `parallelism: 6` because `back-test` is now the job that impacts lead time (since we consider `back-docs` non-blocking, and `back-test` is far slower than all other back jobs).   `parallelism: 8` is 28% faster than `6`, and 4.13% more expensive than `6`.

Which do you prefer? Significantly faster lead time, or slightly more cost?
 I vote for the former, since theoretically there are also multiple cost benefits with improved lead times.
 
<img width="682" alt="Screenshot 2023-03-02 at 11 29 44" src="https://user-images.githubusercontent.com/3944042/222416868-dd9540e7-1aba-42d2-8291-d89ea511a104.png">

Reply:
Speed speed speed, thanks @jinjagit !

Reply:
(just one side-remark, pretty sure CircleCI bills by the started minute, so to be correct you'd need to take the `CEIL` from the time column in your cost calculation. That's likely a little worse for 8x, but that will always fluctuate as soon as we cross minute boundaries on average.

Reply:
Where do find the `CEIL` time?

Reply:
Ahh, I need to round minutes up. Also a good point.

Reply:
OK, subtracting start up time & then rounding up to full minute:

`parallelism: 8` is 28% faster than `6`, and 6.67% more expensive than 6.

So, I'd still go for speed over slight cost impact.